docs/core-components.md

```diff
@@ -42,13 +42,20 @@ The framework provides a number of default connections.
 * `local`: implements a local node.
 * `oef`: wraps the OEF SDK.
 
+<!-- other connections? gym? -->
+
 
 ### Skill
 
-Skills deliver economic value to the AEA by allowing agents to encapsulate and call any kind of code. A skill encapsulates implementations of the abstract base classes `Handler`, `Behaviour`, and `Task`.
+<!-- Skills are the manifestation of the framework's extensibility. 
+A skill is an atomic capability that suggests enables the agent perform a specific  delivers economic value by allowing the agent encapsulate and call any kind of code. A skill encapsulates implementations of the abstract base classes `Handler`, `Behaviour`, and `Task`.
+--> 
+
+<!-- From this, it is really not clear what skills do, and what their purpose is, what the idea behind them is, etc. --> 
+Skills deliver economic value to the AEA by letting the agent call any kind of code. A skill encapsulates implementations of the abstract base classes `Handler`, `Behaviour`, and `Task`.
 
-* `Handler`: each skill has none, one or more `Handler` objects responsible for the registered protocol messaging. Handlers implement reactive behaviour. By understanding the requirements contained in an `Envelope`, the `Handler` reacts appropriately to message requests. Each `Handler` is responsible for one and only one protocol.
-* `Behaviour`: none, one or more `Behaviours` encapsulate actions that cause interactions with other agents initiated by the agent. Behaviours implement proactive behaviour.
+* `Handler`: each skill has none, one or more `Handler` objects, each responsible for the registered messaging protocol. Handlers implement agents' reactive behaviour. By understanding the requirements <!-- which requirements?, where in the envelope?--> contained in an `Envelope`, the `Handler` reacts appropriately to the received messages. Each `Handler` is responsible for only one protocol.
```

Comment:
You could change it to: `If the agent understands the protocol referenced in the `Envelop`, the ..